api/src/main/java/com/codeforcommunity/rest/IpThrottlingFilter.java

```diff
@@ -0,0 +1,41 @@
+package com.codeforcommunity.rest;
+
+import static com.codeforcommunity.rest.ApiRouter.end;
+
+import io.github.bucket4j.Bandwidth;
+import io.github.bucket4j.Bucket;
+import io.github.bucket4j.BucketConfiguration;
+import io.github.bucket4j.TokensInheritanceStrategy;
+import io.vertx.core.Handler;
+import io.vertx.ext.web.RoutingContext;
+
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+
+public class IpThrottlingFilter implements Handler<RoutingContext> {
+  private final BucketConfiguration config;
+  private Map<String, Bucket> buckets;
+
+  public IpThrottlingFilter(BucketConfiguration config) {
+    this.config = config;
+    this.buckets = new HashMap<>();
+  }
+
+  @Override
+  public void handle(RoutingContext ctx) {
+    String ip = ctx.request().remoteAddress().toString();
+    Bucket bucket = this.buckets.get(ip);
+
+    if (bucket == null) {
+      bucket = Bucket.builder().addLimit(this.config.getBandwidths()[0]).build();
+      this.buckets.put(ip, bucket);
+    }
```

Comment:
💭 Is there a config option to expire and delete buckets after a period of inactivity? There's a potential to log so many unique IPs that the bucket map consumes all the available memory. I don't think it's likely since that'd require a huge amount of unique IPs to do so - if it's not a lot of work, feel free to implement it, otherwise don

Reply:
good idea. will do.